File: 3581-the-two-sneaky-numbers-of-digitville/the-two-sneaky-numbers-of-digitville.java
class Solution {
    public int[] getSneakyNumbers(int[] nums) {
        int n = nums.length;
        Set<Integer> seen = new HashSet<>();

        int[] ans = new int[2]; int idx = 0;
        for (int i = 0; i < n; i++) {
            if (!seen.add(nums[i])) {
                ans[idx++] = nums[i];
            }
        }

        return ans;
    }
}

File: 3581-the-two-sneaky-numbers-of-digitville/the-two-sneaky-numbers-of-digitville.cpp
class Solution {
public:
    vector<int> getSneakyNumbers(vector<int>& nums) {
        int n = nums.size();
        unordered_set<int> seen;

        vector<int> ans;
        for (int i = 0; i < n; i++) {
            if (!seen.insert(nums[i]).second) {
                ans.push_back(nums[i]);
            }
        }

        return ans;
    }
};